src/puppetlabs/puppetdb/command.clj

```diff
@@ -428,20 +414,27 @@
         (recur (dec n))
         result))))
 
-(defn process-command-and-respond! [cmd db response-pub-chan stats-atom]
+(defn process-command-and-respond! [{:keys [callback] :as cmd} db response-pub-chan stats-atom]
   (try
     (let [result (call-with-quick-retry 4
                   (fn []
                     (process-command! cmd db)))]
       (swap! stats-atom update :executed-commands inc)
+      (callback {:command cmd :result result})
```

Comment:
This callback is the one attached to the command ref, for use on our "blocking post" feature